server/src/main/resources/rules/rules.js

```diff
@@ -517,7 +527,8 @@ function get_pool_priority(pool, consumer) {
         'roles': 2800,
         'addons': 1400,
         'support_level': 700,
-        'usage': 350
+        'usage': 350,
+        'support_type': 340
```

Comment:
Even though 340 is lower than the weight of usage (350), and higher than the next most powerful weight (150 for the requires_host attribute), and it will work for most use cases, it will not work for a number of use cases.

As the comment above explains, the relationship between each attribute's previous/next weight is important. More specifically: each weight should be larger than all the other attributes below it combined.
In this case this rule is broken because usage (350) will now be less than everything below it combined (service_type+requires_host+virt_only+sockets+ram+cores+vcpu: 670).

Here is an example use case that this will break (as an AutobindRulesTest.java unit test):
<details>
<summary>broken use case</summary>


```
    /*
     * This test demonstrates that a pool that satisfies the consumer's Usage should be selected
     * during autoattach even if all other attributes below it (service_type, sockets, cores & ram, etc.) mismatch,
     * instead of a pool that has a matching service_type, but no other mismatches.
     */
    @SuppressWarnings("checkstyle:localvariablename")
    @Test
    public void testShouldSelectPoolWhenUsageMatchesButServiceTypeSocketsRamCoresMismatch() {
        Product product69 = new Product();
        product69.setId("compliant-69");

        // Consumer specified attributes:
        consumer.setUsage("myusage");
        consumer.setServiceType("myservicetype");
        consumer.setFact("cpu.cpu_socket(s)", "1");
        consumer.setFact("cpu.core(s)_per_socket", "1");
        consumer.setFact("memory.memtotal", "9980456");
        ConsumerInstalledProduct consumerInstalledProduct =
            new ConsumerInstalledProduct(product69);
        consumer.addInstalledProduct(consumerInstalledProduct);

        // --- No satisfied syspurpose attributes on the consumer ---

        // Candidate pools:

        // This pool covers the system's usage attribute, but most other attributes below usage (service_type,
        // cores, ram, etc.) are mismatched.
        Product prodMCT1650 = createSysPurposeProduct(null, null, null,
            null, "myusage", "random_service_type");
        // Set large sockets/cores/ram values to this SKU, to make sure it's priority is reduced compared to
        // the other pool that uses the default priorities for these attributes (prioritization points for
        // cores/sockets/ram works this way because we do not want to to attach a pool that wastes extra
        // resource attributes if we can).
        prodMCT1650.setAttribute(Product.Attributes.SOCKETS, "32");
        prodMCT1650.setAttribute(Product.Attributes.CORES, "32");
        prodMCT1650.setAttribute(Product.Attributes.RAM, "19960912");
        prodMCT1650.addProvidedProduct(product69);
        Pool MCT1650 = TestUtil.createPool(owner, prodMCT1650);
        MCT1650.setId("MCT1650");
        MCT1650.setQuantity(1L);

        // This pool only covers the service_type of the system (and not the usage attribute), but has no
        // other mismatches compared to the previous pool.
        Product genericProduct = createSysPurposeProduct(null, null, null, null, null, "myservicetype");
        genericProduct.addProvidedProduct(product69);
        Pool genericPool = TestUtil.createPool(owner, genericProduct);
        genericPool.setId("genericPool");
        genericPool.setQuantity(1L);

        List<Pool> pools = new ArrayList<>();
        pools.add(MCT1650);
        pools.add(genericPool);

        List<PoolQuantity> bestPools = autobindRules.selectBestPools(consumer,
            new String[]{"compliant-69"}, pools, compliance, null, new HashSet<>(), false);

        assertEquals(1, bestPools.size());
        // We expect MCT1650 to be attached, because the fact that it covers the usage attribute, which is
        // more important than the service_type attribute, should be enough to prioritize it over
        // genericPool, even if other mismatches of attributes of lower importance are in play.
        assertTrue(bestPools.contains(new PoolQuantity(MCT1650, 1)));
    }

```
</details>

To solve this, I would suggest using 350 as the weight for the support_type attribute, and simply double the weight of every attribute above it. So like this:
```
        'products': 11200,
        'roles': 5600,
        'addons': 2800,
        'support_level': 1400,
        'usage': 700,
        'support_type': 350
```

Reply:
Also, if we are going to apply this change, we would also have to change the baseline priority that every pool starts with above from:
 `var priority = 545;` to `var priority =  1105;` (to make sure, as the comment above that explains, we never get priorities below zero, when a pool accrues too many mismatches that result in negative points).

And perhaps we need to change those 2 comments that explain the algorithm to reflect these changes.

Reply:
Adding a comment here for posterity to note the outcome of more discussion between me and @sonalidhome around the adjustments to the algorithm: In addition to the above changes:
- We found out there is a special case around virt_only (see #2388) for the algorithm, whose weight (+100) need to be more powerfull than all the syspurpose attributes' null rule scores combined. The current changes break that, which lead to having to reduce the null rule modifier to half, from `0.01` to `0.005`.
- The current changes also break another goal of the algorithm, which is that: A given syspurpose attribute match should be more powerful than all of the other attribute mismatches combined (even for the mismatches of attributes that are of higher importance). In order to solve that, we decided to also reduce the mismatch rule modifier to half, from `-0.05` to `-0.025`.

Reply:
Ack. Changes are updated.